clang error: No matching function for call to 'hash_combine'

Compiling with Apple clang gives the following error:

```
nvrhi/src/vulkan/vulkan-resource-bindings.cpp:456:17: error: no matching function for call to 'hash_combine'
                nvrhi::hash_combine(viewInfoHash, range.byteOffset);
                ^~~~~~~~~~~~~~~~~~~
In file included from .../nvrhi/src/vulkan/vulkan-resource-bindings.cpp:23:
In file included from .../nvrhi/src/vulkan/vulkan-backend.h:25:
In file included from .../nvrhi/include/nvrhi/vulkan.h:26:
.../nvrhi/include/nvrhi/nvrhi.h:2768:10: note: candidate function template not viable: no known conversion from 'uint64_t' (aka 'unsigned long long') to 'size_t &' (aka 'unsigned long &') for 1st argument
    void hash_combine(size_t& seed, const T& v)
```

And the following warning:

```
.../nvrhi/src/validation/validation-device.cpp:1182:87: warning: '&&' within '||' [-Wlogical-op-parentheses]
        if (!(graphicsApi == GraphicsAPI::D3D12 || graphicsApi == GraphicsAPI::VULKAN && desc.registerSpaceIsDescriptorSet))
                                                ~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.../nvrhi/src/validation/validation-device.cpp:1182:87: note: place parentheses around the '&&' expression to silence this warning
        if (!(graphicsApi == GraphicsAPI::D3D12 || graphicsApi == GraphicsAPI::VULKAN && desc.registerSpaceIsDescriptorSet))
                                                                                      ^
                                                   (                                                                      )
```

I will submit a PR with fixes.

--- a/src/validation/validation-device.cpp
+++ b/src/validation/validation-device.cpp
@@ -1179,7 +1179,7 @@ namespace nvrhi::validation
         }
 
         const GraphicsAPI graphicsApi = m_Device->getGraphicsAPI();
-        if (!(graphicsApi == GraphicsAPI::D3D12 || graphicsApi == GraphicsAPI::VULKAN && desc.registerSpaceIsDescriptorSet))
+        if (!(graphicsApi == GraphicsAPI::D3D12 || (graphicsApi == GraphicsAPI::VULKAN && desc.registerSpaceIsDescriptorSet)))
         {
             if (desc.registerSpace != 0)
             {

--- a/src/vulkan/vulkan-resource-bindings.cpp
+++ b/src/vulkan/vulkan-resource-bindings.cpp
@@ -452,7 +452,7 @@ namespace nvrhi::vulkan
                 auto vkformat = nvrhi::vulkan::convertFormat(format);
                 const auto range = binding.range.resolve(buffer->desc);
 
-                uint64_t viewInfoHash = 0;
+                size_t viewInfoHash = 0;
                 nvrhi::hash_combine(viewInfoHash, range.byteOffset);
                 nvrhi::hash_combine(viewInfoHash, range.byteSize);
                 nvrhi::hash_combine(viewInfoHash, (uint64_t)vkformat);
@@ -771,7 +771,7 @@ namespace nvrhi::vulkan
                     auto vkformat = nvrhi::vulkan::convertFormat(binding.format);
 
                     const auto range = binding.range.resolve(buffer->desc);
-                    uint64_t viewInfoHash = 0;
+                    size_t viewInfoHash = 0;
                     nvrhi::hash_combine(viewInfoHash, range.byteOffset);
                     nvrhi::hash_combine(viewInfoHash, range.byteSize);
                     nvrhi::hash_combine(viewInfoHash, (uint64_t)vkformat);

